public/js/controllers/posts.js

```diff
@@ -1,5 +1,5 @@
 /*global flapperNews */
-flapperNews.controller('MainCtrl', [
+flapperNews.controller('MainController', [
   '$scope',
   'posts',
   function ($scope, posts) {
```

Comment:
This is an opinionated thing so take it with a grain of salt, but it can also be nice to define controllers separately as named constructors. That way they wont show up as anonymous functions in your stack traces and you'll easily know where the issue is. The benefit too, is that you can define them outside of the controller declaration and then attach them later. Which for me helps a lot, because don't have to deal with needless amounts of indentation. Also by defining the controllers as constructors it makes you think in a class oriented way that the `controllerAs` pattern in the templates requires. This is definitely opinionated as many people choose to use anonymous functions, but its something I've grown to like a lot. 

An example would be:

```javascript
'use strict';

// define the constructor
function MainController ($scope, posts) {
  this.posts = posts.posts;
  this.addPost = function () {
     if (!this.title || this.title === '') { return; }
     this.posts.push({
         title: this.title,
         link: this.link,
         upvotes: 0
     });
     this.title = '';
     this.link = '';
  };
  this.incrementUpvotes = function (post) {
     post.upvotes += 1;
  };
}

// then attach
angular.module('flapperNews')
  .controller(['$scope', 'posts', MainController])
;
````